manifests/service.pp

```diff
@@ -11,21 +11,23 @@
 # Sample Usage:
 #
 class activemq::service(
-  $ensure,
-  $service_enable = $::activemq::params::service_enable
+  $ensure         = $activemq::ensure,
+  $service_enable = $activemq::params::service_enable,
 ) {
 
   # Arrays cannot take anonymous arrays in Puppet 2.6.8
   $v_ensure = [ '^running$', '^stopped$' ]
   validate_re($ensure, $v_ensure)
+  $enable = $ensure ? {
+    'running' => true,
+    default   => false,
+  }
 
   validate_bool($service_enable)
 
-  $ensure_real = $ensure
-
   service { 'activemq':
-    enable     => $service_enable,
-    ensure     => $ensure_real,
+    ensure     => $ensure,
+    enable     => $enable,
```

Comment:
This introduce a change in behavior and links the ensure argument with the fact that the service is active or not (which wasn't the case before). Service can be present but disabled.

Moreover with this modification the `$service_enable` param doesn't make sense anymore